### PR TITLE
PLANET-7629 Move Covers block to the master theme

### PR DESCRIPTION
**Pull request by mleray**
### Description

See [PLANET-7629](https://jira.greenpeace.org/browse/PLANET-7629)

It has been removed from the plugin at the same time ([#1253](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/1253)). For now I kept the Campaigns style, I think it would be easier to remove it in a follow-up PR (and after the [migration](https://github.com/greenpeace/planet4-master-theme/pull/2349) has been merged).

### Testing

The Covers block should behave as expected, both new and existing blocks with the different styles. [This page](https://www-dev.greenpeace.org/test-tavros/all-the-covers/) for example has all of them so you can use it for testing!

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [tavros](https://www-dev.greenpeace.org/test-tavros) | [admin](https://www-dev.greenpeace.org/test-tavros/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-tavros/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-tavros) | [composer-local.json](https://github.com/greenpeace/planet4-test-tavros/blob/main/composer-local.json)

:watch: 2024.10.10 09:03:42

File: assets/src/blocks/Covers/CoversEditorScript.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import {registerCoversBlock} from './CoversBlock';
+
+registerCoversBlock();
```
Comment (mardelnet) [inline assets/src/blocks/Covers/CoversEditorScript.js:3]: Can't we just move the code of the `registerCoversBlock` function to this file?

File: src/Loader.php [hunk 2]
```diff
@@ -211,6 +211,7 @@ public static function add_blocks(): void
         new Blocks\TableOfContents();//NOSONAR
         new Blocks\TakeActionBoxout();//NOSONAR
         new Blocks\Timeline();//NOSONAR
+        new Blocks\Covers();//NOSONAR
```
Comment (mardelnet) [inline src/Loader.php:214]: This is super nicky picky, but it is not alphabetically ordered 😆

File: functions.php
Comment (mardelnet) [inline functions.php:1]: This is an idea for another PR:
It seems to me that such an important piece of code like this one in charge of registering the API endpoints should be part of a separate class and not just a hook in the middle of the `functions.php` file.

**Review by mardelnet — COMMENTED** [review]
Nice job, @mleray 
Let me give you some initial comments while I complete the review:

The following elements were not brought from the plugin, don't we need them?
- public/images/climate/covers-block.svg
- public/images/oil/covers-block.svg
- tests/e2e/covers-campaign.spec.js

File: assets/src/blocks/Covers/CoversEditorScript.js (hunk 1 above)
Comment (mardelnet) [reply assets/src/blocks/Covers/CoversEditorScript.js:3]: Forget this comment, it does not make sense to introduce changes to a block that will be migrated and removed soon.

File: functions.php [hunk 3]
```diff
@@ -102,6 +102,7 @@ function (): void {
         Api\Search::register_endpoint();
         Api\Settings::register_endpoint();
         Api\ENForm::register_endpoint();
+        Api\Covers::register_endpoint();
```
Comment (mardelnet) [inline functions.php:105]: This is an idea for another PR:
It seems to me that such an important piece of code like this one in charge of registering the API endpoints should be part of a separate class and not just a hook in the middle of the `functions.php` file.

**Comment by mleray** [comment]
> The following elements were not brought from the plugin, don't we need them?
> 
> * public/images/climate/covers-block.svg
> * public/images/oil/covers-block.svg
> * tests/e2e/covers-campaign.spec.js

The first two are not used anymore, so no. The third one I didn't move because at first I thought I would also retire the Campaigns style in this PR, but since now I think it would be best to handle in a follow-up PR I could bring it back... But also it will be removed soon anyway, so what do you think?

File: src/Loader.php (hunk 2 above)
Comment (mleray) [reply src/Loader.php]: Not nit picky, we should keep it ordered! Updated 👍

**Comment by mardelnet** [comment]
> The first two are not used anymore, so no we don't need them. The third one I didn't move because at first I thought I would also retire the Campaigns style in this PR, but since now I think it would be best to handle in a follow-up PR I could bring it back... But also it will be removed soon anyway, so what do you think?

Let's bring it now so we don't forget about it later.

**Review by mardelnet — APPROVED** [review]
LGTM! Good job, @mleray

File: functions.php (hunk 3 above)
Comment (mardelnet) [reply functions.php:105]: No comments here so I assume you disagree 😆  That's okay

Comment (mleray) [reply functions.php:105]: Sorry no I don't disagree haha! I don't have a strong opinion either way.

**Pull request merged**